Makefile

```diff
@@ -1,9 +1,14 @@
 # Do things in edx-platform
+.PHONY: clean extract_translations pull_translations push_translations requirements upgrade
 
 # Careful with mktemp syntax: it has to work on Mac and Ubuntu, which have differences.
 PRIVATE_FILES := $(shell mktemp -u /tmp/private_files.XXXXXX)
 
-clean:
+help: ## display this help message
+	@echo "Please use \`make <target>' where <target> is one of"
+	@perl -nle'print $& if m{^[a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
```

Comment:
How can it be that make doesn't provide this functionality natively?? :(

Reply:
Just because, I tried my hand at a python replacement.  It's longer...
```
	@python -c 'import fileinput,re; ms=filter(None, (re.search("([a-zA-Z_-]+):.*?## (.*)$$",l) for l in fileinput.input())); print("\n".join(sorted("\033[36m  {:25}\033[0m {}".format(*m.groups()) for m in ms)))' $(MAKEFILE_LIST)
```
But how did perl get in there in the first place? It's just a grep:
```
	@grep -E '^[a-zA-Z_-]+:.*?##' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
```
Which is shorter than the original, and just to continue the golfing:
```
	@grep '^[a-zA-Z]' $(MAKEFILE_LIST) | sort | awk -F ':.*?## ' 'NF==2 {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
```

Reply:
BTW, you should add "help" to the .PHONY target, I think.

Reply:
Fingers crossed! :) https://github.com/pydanny/cookiecutter-djangopackage/pull/378